plotly/offline/offline.py

```diff
@@ -50,26 +46,36 @@ def get_plotlyjs():
 
 def init_notebook_mode():
     """
-    Initialize Plotly Offline mode in an IPython Notebook.
-    Run this function at the start of an IPython notebook
-    to load the necessary javascript files for creating
-    Plotly graphs with plotly.offline.iplot.
+    Initialize plotly.js in the browser if it hasn't been loaded into the DOM
+    yet. This is an idempotent method and can and should be called from any
+    offline methods that require plotly.js to be loaded into the notebook dom.
     """
-    if not tools._ipython_imported:
+    warnings.warn('''
+        `init_notebook_mode` is deprecated and will be removed in the
+        next release. Notebook mode is now automatically initialized when
+        notebook methods are invoked, so it is no
+        longer necessary to manually initialize.
+    ''', DeprecationWarning)
+
+    if not _ipython_imported:
         raise ImportError('`iplot` can only run inside an IPython Notebook.')
-    from IPython.display import HTML, display
 
-    global __PLOTLY_OFFLINE_INITIALIZED
-    if not __PLOTLY_OFFLINE_INITIALIZED:
-        display(HTML("<script type='text/javascript'>" +
-                     "define('plotly', function(require, exports, module) {" +
-                     get_plotlyjs() +
-                     "});" +
-                     "require(['plotly'], function(Plotly) {" +
-                     "window.Plotly = Plotly;" +
-                     "});" +
-                     "</script>"))
-    __PLOTLY_OFFLINE_INITIALIZED = True
+    script_inject = (
+        ''
+        '<script type=\'text/javascript\'>'
+        'if(!window.Plotly){{'
+        'define(\'plotly\', function(require, exports, module) {{'
+        '{script}'
+        '}});'
+        'require([\'plotly\'], function(Plotly) {{'
+        'console.log(Plotly);'
+        'window.Plotly = Plotly;'
+        '}});'
+        '}}'
+        '</script>'
+        '').format(script=get_plotlyjs())
+
+    display(HTML(script_inject))
```

Comment:
:cow2: could use less escaping by following original formatting, but not needed!